bin/pre-commit-hook.js

```diff
@@ -28,3 +32,28 @@ if ( lintResult.status ) {
 		'repeat the commit command with --no-verify to avoid this check.' );
 	process.exit( 1 );
 }
+
+
+// run prettier for any files in the commit that have @format within their first docblock
+const formatRegex = /@format/;
+const firstDocBlockStart = /\/\*\*/;
+const firstDocBlockEnd = /\*\//;
+
+files
+	.map( file => path.join( __dirname, '../', file ) )
+	.forEach( file => {
+		fs.readFile( file, 'utf8', ( err, data ) => {
+			const firstDocBlockStartIndex = _.get( firstDocBlockStart.exec( data ), 'index' );
+			const firstDocBlockEndIndex = _.get( firstDocBlockEnd.exec( data ), 'index' );
+			const firstDocBlockText =
+				_.isInteger( firstDocBlockStartIndex ) &&
+				_.isInteger( firstDocBlockEndIndex ) &&
+				data.substring( firstDocBlockStartIndex, firstDocBlockEndIndex + 2 );
+
+			if ( firstDocBlockText && formatRegex.test( firstDocBlockText ) ) {
+				const formattedData = prettier.format( data, {} );
+				fs.writeFileSync( file, formattedData );
+				execSync( `git add ${ file }` );
+			}
```

Comment:
from my testing, its a premature optimization. but I'll throw it in because this affects everyone.